### PR TITLE
Deploying an SPA to Google Cloud

### PR DESCRIPTION
**Pull request by samcdavid**
Why
---

- The documentation for hosting a static site on GCP, particularly with
our setup, is not fully documented anywhere. It would be helpful to have
these steps documented so we can easily repeat this process again in the
future.

This change addresses the need by
---------------------------------

- Document the build process for an SPA to be deployed
- Document the setup process for creating the necessary resources in GCP
- Document the DNS changes that need to be made for accessing the SPA
- Document a scripted deployment for deploying the SPA.

### CODE REVIEW

File: content/post/deploy-spa-to-google-cloud.md [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+In my case, the start script is used to run my webpack-dev-server and uses the
+development webpack config. The build script runs the production webpack config
+and stores the output in the dist folder.
+
+## Setting up the Infrastructure
+
+When setting up your infrastructure you will need to do two things: configure
+your DNS and create the storage bucket. We will configure the DNS first so there
+is proper time for it to propagate.
+
```
Comment (bbrock25) [inline content/post/deploy-spa-to-google-cloud.md]: could there be value in including a screenshot or three here?

Comment (bbrock25) [reply content/post/deploy-spa-to-google-cloud.md]: also, does gcp have an equivalent command line utility to aws cli?

File: content/post/deploy-spa-to-google-cloud.md [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+the directions for domain ownership verification.
+
+Now that you're bucket is created, you are ready to run your production build
+and upload the files to your bucket.
+
+## First Deployment
+
+For the first deployment, we are going to do everything manually. Begin by
+opening your terminal to your SPA and run `yarn build`. Once the build process
+is complete, go back to the GCP console and upload the files to your bucket. If
```
Comment (bbrock25) [inline content/post/deploy-spa-to-google-cloud.md]: upload which files?

File: content/post/deploy-spa-to-google-cloud.md [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+with a scripted deployment.
+
+## Further Considerations
+
+Now that you know how to build and deploy your SPA to a GCP Bucket you may think
+that you're job is done. However, there are many other details for you to
+consider. One of the technical limitations of hosting your SPA on a GCP Bucket
+is your SPA is only accessible via HTTP. In order to protect your users you will
+need to setup a load balancer so you can serve your users via HTTPS (which will
+be covered in another blog post).
```
Comment (bbrock25) [inline content/post/deploy-spa-to-google-cloud.md]: maybe link to the google docs here to get folks started in the right direction in the meantime.

File: content/post/deploy-spa-to-google-cloud.md (hunk 1 above)
Comment (samcdavid) [reply content/post/deploy-spa-to-google-cloud.md]: There is an sdk you can install that has some cli tools in it. I mention installed it later on in the post.

I can dig up some screen shots.

File: content/post/deploy-spa-to-google-cloud.md (hunk 2 above)
Comment (samcdavid) [reply content/post/deploy-spa-to-google-cloud.md]: I am changing the wording from files to build output. The next sentence tells them what the build output files are if they are using the sample repo.

File: content/post/deploy-spa-to-google-cloud.md (hunk 3 above)
Comment (samcdavid) [reply content/post/deploy-spa-to-google-cloud.md]: adding the link now

**Pull request closed without merging**